### PR TITLE
[framework] admin can manage customer user group roles

### PR DESCRIPTION
**Pull request by malyMiso**
#### Description, the reason for the PR
admin can manage customer user group roles

#### Fixes issues
... 

#### Check the appropriate checkboxes below, please

- [ ] [BC breaks](https://docs.shopsys.com/en/latest/contributing/backward-compatibility-promise/) 
- [x] New feature 
- [x] I have read and signed [License Agreement for contributions](https://www.shopsys.com/license-agreement)

----
:globe_with_meridians: Live Preview:
  - https://mt-roles-admin.odin.shopsys.cloud
  - https://cz.mt-roles-admin.odin.shopsys.cloud

### CODE REVIEW

File: packages/framework/src/Model/Customer/User/Role/CustomerUserRoleGroupRepository.php [hunk 1]
```diff
@@ -61,4 +63,36 @@ public function getByUuid(string $uuid): CustomerUserRoleGroup
...
+        return $this->getCustomerUserRoleGroupRepository()->createQueryBuilder('cug');
+    }
+
+    /**
+     * @param string $locale
+     * @return \Doctrine\ORM\QueryBuilder
+     */
+    public function getAllQueryBuilderByLocale(string $locale): QueryBuilder
+    {
+        $queryBuilder = $this->getCustomerUserRoleGroupRepository()->createQueryBuilder('cug');
```
Comment (grossmannmartin) [inline packages/framework/src/Model/Customer/User/Role/CustomerUserRoleGroupRepository.php]: _nitpick_ 

```suggestion
        $queryBuilder = $this->getAllQueryBuilder();
```

File: packages/framework/src/Controller/Admin/CustomerUserRoleGroupController.php [hunk 2]
```diff
@@ -30,4 +38,38 @@ public function listAction(): Response
...
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            $customerUserRoleGroup = $this->customerUserRoleGroupFacade->create($roleGroupData);
+
+            $this->addSuccessFlashTwig(
+                t('Customer user role group <strong><a href="{{ url }}">{{ name }}</a></strong> was created'),
+                [
+                    'name' => $customerUserRoleGroup->getName(),
+                    'url' => $this->generateUrl('admin_superadmin_customer_user_role_group_list', ['id' => $customerUserRoleGroup->getId()]),
```
Comment (grossmannmartin) [inline packages/framework/src/Controller/Admin/CustomerUserRoleGroupController.php]:
```suggestion
                    'url' => $this->generateUrl('admin_superadmin_customer_user_role_group_edit', ['id' => $customerUserRoleGroup->getId()]),
```

File: packages/framework/src/Form/Admin/Customer/RoleGroup/CustomerUserRoleGroupFormType.php [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+     * @param array $options
+     */
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        $builder->add('names', LocalizedType::class, [
+            'required' => true,
+            'label' => t('Role name'),
+            'entry_options' => [
+                'constraints' => [
+                    new Constraints\NotBlank(['message' => 'Please enter order name']),
```
Comment (grossmannmartin) [inline packages/framework/src/Form/Admin/Customer/RoleGroup/CustomerUserRoleGroupFormType.php]:
```suggestion
                    new Constraints\NotBlank(['message' => 'Please enter role name']),
```

File: packages/framework/src/Model/Customer/User/Role/CustomerUserRole.php [hunk 4]
```diff
@@ -9,4 +9,15 @@ class CustomerUserRole
     public const ROLE_API_LOGGED_CUSTOMER = 'ROLE_API_LOGGED_CUSTOMER';
     public const ROLE_API_ALL = 'ROLE_API_ALL';
     public const ROLE_API_CUSTOMER_SELF_MANAGE = 'ROLE_API_CUSTOMER_SELF_MANAGE';
+
+    /**
+     * @return array<string, string>
```
Comment (grossmannmartin) [inline packages/framework/src/Model/Customer/User/Role/CustomerUserRole.php:14]: does it make sense to extend the `Shopsys\FrameworkBundle\Component\Enum\AbstractEnum` for enum-like behavior?

File: packages/framework/src/Model/Customer/User/Role/CustomerUserRoleGroup.php [hunk 5]
```diff
@@ -51,8 +51,24 @@ class CustomerUserRoleGroup extends AbstractTranslatableEntity
     public function __construct(CustomerUserRoleGroupData $customerUserRoleGroupData)
     {
         $this->translations = new ArrayCollection();
-        $this->roles = $customerUserRoleGroupData->roles;
         $this->uuid = $customerUserRoleGroupData->uuid ?: Uuid::uuid4()->toString();
+        $this->setData($customerUserRoleGroupData);
+    }
```
Comment (grossmannmartin) [inline packages/framework/src/Model/Customer/User/Role/CustomerUserRoleGroup.php:53]: are the constants DEFAULT_ROLE_GROUP_NAME and USER_ROLE_GROUP_NAME in this file used anywhere?

File: packages/framework/src/Model/Customer/User/Role/CustomerUserRoleGroupGridFactory.php [hunk 6]
```diff
@@ -41,6 +41,8 @@ public function create(): Grid
 
         $grid->setActionColumnClassAttribute('table-col table-col-10');
         $grid->addEditActionColumn('admin_superadmin_customer_user_role_group_edit', ['id' => 'cug.id']);
+        $grid->addDeleteActionColumn('admin_superadmin_customer_user_role_group_delete', ['id' => 'cug.id'])
+            ->setConfirmMessage(t('Do you really want to remove this administrator role group?'));
```
Comment (grossmannmartin) [inline packages/framework/src/Model/Customer/User/Role/CustomerUserRoleGroupGridFactory.php]:
```suggestion
            ->setConfirmMessage(t('Do you really want to remove this customer user role group?'));
```

File: project-base/app/tests/App/Smoke/Http/RouteConfigCustomization.php [hunk 7]
```diff
@@ -111,6 +111,9 @@ private function filterRoutesForTesting(RouteConfigCustomizer $routeConfigCustom
             })
             ->customizeByRouteName('admin_flag_delete', function (RouteConfig $config) {
                 $config->skipRoute('Deletion of flag from ShopSys is disabled.');
+            })
+            ->customizeByRouteName('admin_superadmin_customer_user_role_group_delete', function (RouteConfig $config) {
+                $config->skipRoute('All customer user role groups are assigned to demo customer users.');
```
Comment (grossmannmartin) [inline project-base/app/tests/App/Smoke/Http/RouteConfigCustomization.php]: but even in that case it should return the 200 response, or?

File: packages/framework/src/Resources/translations/messages.cs.po [hunk 8]
```diff
@@ -751,6 +754,9 @@ msgstr "Vytvořit nový stát"
 msgid "Create new customer"
 msgstr "Vytvořit nového zákazníka"
 
+msgid "Create new customer user role group"
+msgstr "Vytvořte novou skupinu uživatelských práv"
```
Comment (grossmannmartin) [inline packages/framework/src/Resources/translations/messages.cs.po]:
```suggestion
msgstr "Vytvořit novou skupinu uživatelských práv"
```

File: packages/framework/src/Resources/translations/messages.cs.po [hunk 9]
```diff
@@ -1297,6 +1321,12 @@ msgstr "Editace zákazníka"
 msgid "Editing customer - %name%"
 msgstr "Editace zákazníka - %name%"
 
+msgid "Editing customer user role group"
+msgstr "Editace skupiny práv zákazníka"
```
Comment (grossmannmartin) [inline packages/framework/src/Resources/translations/messages.cs.po]: we should unify the translation of customer user role group. I see two different translations: `skupina práv zákazníka`, `skupina uživatelských práv`, and `skupina práv`

**Review by grossmannmartin — CHANGES_REQUESTED** [review]

File: packages/framework/src/Model/Customer/User/Role/CustomerUserRole.php (hunk 4 above)
Comment (malyMiso) [reply packages/framework/src/Model/Customer/User/Role/CustomerUserRole.php:14]: I do not see any advantage of extending AbstractEnum. What shall I use it for ?

File: packages/framework/src/Model/Customer/User/Role/CustomerUserRoleGroupRepository.php (hunk 1 above)
Comment (malyMiso) [reply packages/framework/src/Model/Customer/User/Role/CustomerUserRoleGroupRepository.php]: fixed: https://github.com/shopsys/shopsys/pull/3323/commits/63f3ee1a735158a4ff936778fa1e632cf6f23710

File: packages/framework/src/Controller/Admin/CustomerUserRoleGroupController.php (hunk 2 above)
Comment (malyMiso) [reply packages/framework/src/Controller/Admin/CustomerUserRoleGroupController.php]: fixed: https://github.com/shopsys/shopsys/pull/3323/commits/8af81e41f34a51ca1bd052e8626df874b19a5454

File: packages/framework/src/Form/Admin/Customer/RoleGroup/CustomerUserRoleGroupFormType.php (hunk 3 above)
Comment (malyMiso) [reply packages/framework/src/Form/Admin/Customer/RoleGroup/CustomerUserRoleGroupFormType.php]: fixed: https://github.com/shopsys/shopsys/pull/3323/commits/15f1f403773bc92ca968f318298f9f1b714b82e8

File: packages/framework/src/Model/Customer/User/Role/CustomerUserRoleGroup.php (hunk 5 above)
Comment (malyMiso) [reply packages/framework/src/Model/Customer/User/Role/CustomerUserRoleGroup.php:53]: You are right. They are not used. Removed: https://github.com/shopsys/shopsys/pull/3323/commits/12967f3bac2c7511cd32343a460ff3c34b933f44

File: packages/framework/src/Model/Customer/User/Role/CustomerUserRoleGroupGridFactory.php (hunk 6 above)
Comment (malyMiso) [reply packages/framework/src/Model/Customer/User/Role/CustomerUserRoleGroupGridFactory.php]: fixed: https://github.com/shopsys/shopsys/pull/3323/commits/bf6fd94475718ddcc733ade0fd191b68e1328562

File: project-base/app/tests/App/Smoke/Http/RouteConfigCustomization.php (hunk 7 above)
Comment (malyMiso) [reply project-base/app/tests/App/Smoke/Http/RouteConfigCustomization.php]: Thanks for pointing it out. It was a tricky one, but it is fixed now: https://github.com/shopsys/shopsys/pull/3323/commits/356d89d64951e665481113ac0be228d6d8f25821

File: packages/framework/src/Resources/translations/messages.cs.po (hunk 8 above)
Comment (malyMiso) [reply packages/framework/src/Resources/translations/messages.cs.po]: all translations fixed here: https://github.com/shopsys/shopsys/pull/3323/commits/0b690ff3e88fe5ddc262451e058eaba25effc2c2

File: packages/framework/src/Resources/translations/messages.cs.po (hunk 9 above)
Comment (malyMiso) [reply packages/framework/src/Resources/translations/messages.cs.po]: all translations fixed here: https://github.com/shopsys/shopsys/pull/3323/commits/0b690ff3e88fe5ddc262451e058eaba25effc2c2

File: packages/framework/src/Model/Customer/User/Role/CustomerUserRole.php (hunk 4 above)
Comment (grossmannmartin) [reply packages/framework/src/Model/Customer/User/Role/CustomerUserRole.php:14]: We could easily get all cases, but it's not necessary in this scenario

File: packages/framework/src/Resources/translations/validators.en.po [hunk 10]
```diff
@@ -358,6 +358,9 @@ msgstr ""
 msgid "Please enter quantity"
 msgstr ""
 
+msgid "Please enter role name"
+msgstr ""
```
Comment (grossmannmartin) [inline packages/framework/src/Resources/translations/validators.en.po:365]: please fill in a translation here ;)

**Review by grossmannmartin — APPROVED** [review]

File: packages/framework/src/Resources/translations/validators.en.po (hunk 10 above)
Comment (malyMiso) [reply packages/framework/src/Resources/translations/validators.en.po:365]: fixed

File: packages/framework/src/Model/Customer/User/Role/CustomerUserRoleGroup.php [hunk 11]
```diff
@@ -51,8 +48,24 @@ class CustomerUserRoleGroup extends AbstractTranslatableEntity
...
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Customer\User\Role\CustomerUserRoleGroupData $administratorRoleGroupData
+     */
+    public function edit(CustomerUserRoleGroupData $administratorRoleGroupData): void
```
Comment (pk16011990) [inline packages/framework/src/Model/Customer/User/Role/CustomerUserRoleGroup.php]: Probably typo. It should be `$customerUserRoleGroupData`

Comment (malyMiso) [reply packages/framework/src/Model/Customer/User/Role/CustomerUserRoleGroup.php]: :+1: fixed in https://github.com/shopsys/shopsys/pull/3323/commits/1b3d0b8c0cd06098e652f49d9f10265a50f855f8

**Pull request merged**